com.unity.render-pipelines.universal/Editor/ShaderPreprocessor.cs

```diff
@@ -153,8 +153,8 @@ bool StripUnusedPass(ShaderFeatures features, ShaderSnippetData snippetData)
 
         bool StripUnusedFeatures(ShaderFeatures features, Shader shader, ShaderSnippetData snippetData, ShaderCompilerData compilerData)
         {
-            var globalSettings = UniversalRenderPipelineGlobalSettings.Ensure();
-            bool stripDebugDisplayShaders = !Debug.isDebugBuild || !globalSettings.supportRuntimeDebugDisplay;
+            var globalSettings = UniversalRenderPipelineGlobalSettings.instance;
```

Comment:
Then, should we do as in HDRP: Checking that the GlobalSettings is here and at last version (if any migration) as a prebuild check ?

Reply:
yes probably right

Reply:
I made [this PR](https://github.com/Unity-Technologies/Graphics/pull/4951) to address this, but I started to wonder whether it's premature. There's no issue currently - there doesn't seem to be migration of URP asset -> URP global settings (yet). Let me know what you think.

Reply:
IMO it should be required if there is migration but without it it is perhaps premature yes. The good news is that it will be hard to broke to as without migration there is almost nothing to test.
I think it is still a good Idea as you encountered an issue previously because it was tried to be created. So checking the existance of at least one seems not premature at the moment.